src/sentry/integrations/vercel/client.py

```diff
@@ -82,28 +82,32 @@ def create_deploy_webhook(self):
         return response
 
     def get_env_vars(self, vercel_project_id):
-        return self.paginate(self.GET_ENV_VAR_URL % vercel_project_id, "envs")
+        return self.get(self.GET_ENV_VAR_URL % vercel_project_id)
 
     def create_secret(self, vercel_project_id, name, value):
         data = {"name": name, "value": value}
         response = self.post(self.SECRETS_URL, data=data)["uid"]
         return response
 
     def create_env_variable(self, vercel_project_id, key, value):
-        data = {"key": key, "value": value, "target": "production"}
-        response = self.post(self.ENV_VAR_URL % vercel_project_id, data=data)
-        return response
-
-    def update_env_variable(self, vercel_project_id, key, value):
+        data = {"key": key, "value": value, "target": ["production"], "type": "secret"}
         try:
-            self.delete(
-                self.DELETE_ENV_VAR_URL % (vercel_project_id, key),
-                allow_text=True,
-                params={"target": "production"},
-            )
+            return self.post(self.ENV_VAR_URL % vercel_project_id, data=data)
         except ApiError as e:
-            # we can ignore 404 errors here since we are just trying to delete
-            if e.code != 404:
-                raise
-
-        return self.create_env_variable(vercel_project_id, key, value)
+            if e.json:
+                if e.json["error"]:
+                    if e.json["error"]["code"] == "ENV_ALREADY_EXISTS":
+                        return self.update_env_variable(vercel_project_id, key, value, data)
```

Comment:
@ceorourke there are cleaner ways of doing this:
`if e.json and e.json.get("error", {}).get("code"):`